### PR TITLE
Change jerry_api_string_to_char_buffer so it doesn't add '\0' to the end of output buffer.

### PR DESCRIPTION
**Pull request by sand1k**
Fixes issue #439.

### CODE REVIEW

File: jerry-core/jerry.cpp [hunk 1]
```diff
@@ -274,10 +274,13 @@ jerry_api_convert_api_value_to_ecma_value (ecma_value_t *out_value_p, /**< out:
...
 /**
- * Copy string characters to specified buffer, append zero character at end of the buffer.
+ * Copy string characters to specified buffer.
+ *
+ * Note:
+ *   '\0' could occur in characters.
  *
  * @return number of bytes, actually copied to the buffer - if string's content was copied successfully;
- *         otherwise (in case size of buffer is insuficcient) - negative number, which is calculated
+ *         otherwise (in case size of buffer is insufficient) - negative number, which is calculated
```
Comment (egavrin) [inline jerry-core/jerry.cpp]: ^_^

**Comment by egavrin** [comment]
@seanshpark will this change affect IoTjs?

**Comment by seanshpark** [comment]
@egavrin , I think the change is ok with iotjs. but with latest jerryscript master (not this change) fails to run iotjs tests so cannot check with tests. have to find the revision of the failure.

**Comment by zherczeg** [comment]
From technical point of view, LGTM.

**Comment by sand1k** [comment]
@seanshpark, did you find the reason of failure?
@egavrin, could I land the patch?

**Comment by egavrin** [comment]
Good to me. I'll land

**Comment by seanshpark** [comment]
@sand1k , `@seanshpark, did you find the reason of failure?`: yes, ILyoan has found and fixed in iotjs.

**Pull request merged**